openapi_toolset/django_plugin/middlewares.py

```diff
@@ -37,24 +37,29 @@ def __call__(self, request):
         content_type = response.get('Content-Type')
         # in case of application/json;charset=UTF-8
         content_type = content_type.split(';', 1)[0]
+        # only check response in json format
+        if not content_type == 'application/json':
+            return response
+
         path = request.path
         method = request.method.lower()
-        api_spec = API_SPEC.get_operation_spec(path, method)
-        if not api_spec:
-            return self.raise_missing_doc(response)
+
+        schema = json_content = schema = None
         try:
-            # only check response in json format
-            if not content_type == 'application/json':
-                return response
-            content = response.content.decode(response.charset)
-            json_content = json.loads(content)
+            api_spec = API_SPEC.get_operation_spec(path, method)
+            if not api_spec:
+                return self.missing_doc_handler(request, response)
             schema = api_spec.get_response_body_schema()
             if not schema:
                 return self.missing_doc_handler(request, response)
+            content = response.content.decode(response.charset)
+            json_content = json.loads(content)
             jsonschema.validate(json_content, schema)
             self.log(request, 'info', 'resp match doc')
             return response
-        except jsonschema.exceptions.ValidationError as err:
+        except Exception as err:
```

Comment:
I think we should catch the exactly exception (such as jsonschema.exceptions._Error) rather than all exceptions.
Any unexpected excepions may be buges and we should fix it instead of ignoring them.

Reply:
You should create CI and tests to catch bugs.

This is other peoples production servers.
Exceptions will be logged.  Hopefully they report them.